src/compiler/transformers/transform-utils.ts

```diff
@@ -761,3 +758,50 @@ export interface ConvertIdentifier {
   __identifier: boolean;
   __escapedText: string;
 }
+
+/**
+ * Helper method for retrieving all decorators & modifiers from a TypeScript {@link ts.Node} entity.
+ *
+ * Starting with TypeScript v4.8, decorators and modifiers have been coalesced into a single field, and retrieving
+ * decorators directly has been deprecated. This helper function pulls all decorators & modifiers out of said field.
+ *
+ * @see {@link https://devblogs.microsoft.com/typescript/announcing-typescript-4-8/#decorators-are-placed-on-modifiers-on-typescripts-syntax-trees|The TypeScript 4.8 Announcement}
+ *
+ * @param node the node to pull decorators & modifiers out of
+ * @returns a list containing decorators & modifiers on the node, otherwise undefined
```

Comment:
This will never return `undefined` tho, right? Since you use nullish coalescing on the returns of `retrieveTsDecorators` and `retrieveTsModifiers`

Reply:
Yeah, good point - I removed this (the ability to return undefined) during refactoring. 36c402f13bface2a5dff9c05fd81fc89d088cb5b removes this part of the comment